components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -153,17 +153,31 @@ init_scorpio(const int atm_id)
 }
 
 void AtmosphereDriver::
-init_time_stamps (const util::TimeStamp& run_t0, const util::TimeStamp& case_t0)
+init_time_stamps (const util::TimeStamp& run_t0, const util::TimeStamp& case_t0, int run_type)
 {
   m_atm_logger->info("  [EAMxx] Run  start time stamp: " + run_t0.to_string());
   m_atm_logger->info("  [EAMxx] Case start time stamp: " + case_t0.to_string());
 
+  EKAT_REQUIRE_MSG (case_t0<=run_t0,
+      "Error! Case t0 time stamp must precede the run t0 time stamp.\n"
+      "  - case t0: " + case_t0.to_string() + "\n"
+      "  - run  t0: " + run_t0.to_string() + "\n");
+
   // Initialize time stamps
   m_run_t0 = m_current_ts = run_t0;
   m_case_t0 = case_t0;
-}
-
 
+  switch (run_type) {
+    case 0:
+      m_run_type = RunType::Initial; break;
```

Comment:
Does this case need to require `case_t0 == run_t0`? Similar for restart, require `case_t0 < run_t0`?

Reply:
I suppose we could check. But I wonder if it matters at this point. The big need for case-vs-run t0 was to detect restarts. Now case_t0 is basically just used only as a "reference" time for output's time coord.

Reply:
Ah, wait. The output mgr still decides whether this is a rest run based on case_t0<run_t0. I'll pass run type to OM as well.